### PR TITLE
chore(graphql-react-apollo): Update to Apollo Client 3

### PR DESCRIPTION
**Pull request by atdrago**
Hi! Hopefully this PR finds you well! I'm currently trying to add MSW to an existing project that uses Apollo Client 3, and I noticed that the documentation needed some updates. Please let me know if there's anything I've missed, but these were the changes I needed to make to my project to get MSW working with Apollo Client 3.

## Changes

This PR updates `examples/graphql-react-apollo` to use Apollo Client 3, the latest version of the Apollo client, and also makes two fixes to the Apollo implementation. 

### Apollo Client 3 Breaking Changes

There are a handful of breaking changes in 3.x, but specifically the following things required updates:

- [All Apollo dependencies now live under `@apollo/client`](https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/#updating-imports)
- [`apollo-boost` is retired and all references to the project should be removed](https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/#apollo-boost)

### Other Fixes

- `cache` is a required property of the [`ApolloClient` constructor](https://github.com/apollographql/apollo-client/blob/19f1febdad7a91a9f27c6bb42f1162c35288fa40/src/core/ApolloClient.ts#L40-L57), but it wasn't being passed. This was causing the app to crash when I tried to run it. I created an `InMemoryCache` to fulfill this requirement.
- `fetch` is not a property of the [`ApolloClient` constructor](https://github.com/apollographql/apollo-client/blob/19f1febdad7a91a9f27c6bb42f1162c35288fa40/src/core/ApolloClient.ts#L40-L57), so we can't pass our `fetch` override in there. [We _can_ pass it into an `HttpLink` though](https://github.com/apollographql/apollo-client/blob/19f1febdad7a91a9f27c6bb42f1162c35288fa40/src/link/http/selectHttpOptionsAndBody.ts#L34), so I made this change to ApolloClient.js.

## Example checklist

I'm not creating an example, but I've tried to follow the guidelines. Since I'm updating an old example, and not adding a new one, I'm assuming all the existing tests are sufficient. Please let me know if you think there's somewhere I could test further.

- [x] **I've followed the [Contribution guidelines](./CONTRIBUTING.md)**.
- [x] I've verified that all ~~added~~ tests pass by running `npm test`. -- _I didn't add any tests, but all existing tests pass_
- [x] I've included only the necessary minimum setup to showcase the example.
- ~~[ ] I've added a `README.md` file to the root directory of my example.~~ -- _I didn't update the existing README since none of the language really needed to change_

### CODE REVIEW

File: examples/graphql-react-apollo/src/ApolloClient.js [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
-// in both application runtime and tests.
-export const client = new ApolloClient({
+const cache = new InMemoryCache()
+
+const link = createHttpLink({
   uri: 'http://localhost:3000/graphql',
 
   // Use explicit `window.fetch` so tha outgoing requests
   // are captured and deferred until the Service Worker is ready.
   fetch: (...args) => fetch(...args),
```
Comment (atdrago) [inline examples/graphql-react-apollo/src/ApolloClient.js:10]: I noticed that changing this to `fetch: window.fetch` also worked, but I didn't want to make unnecessary changes. Just calling it out as a potential simplification
```suggestion
  fetch: window.fetch,
```

File: examples/graphql-react-apollo/package.json [hunk 2]
```diff
@@ -2,11 +2,10 @@
   "name": "graphql-react-apollo",
   "version": "0.1.0",
   "dependencies": {
-    "@apollo/react-hooks": "^3.1.5",
+    "@apollo/client": "^3.2.2",
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
-    "apollo-boost": "^0.4.9",
```
Comment (atdrago) [inline examples/graphql-react-apollo/package.json:9]: All Apollo dependencies now live under `@apollo/client`

File: examples/graphql-react-apollo/src/ApolloClient.js [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
-// in both application runtime and tests.
-export const client = new ApolloClient({
+const cache = new InMemoryCache()
+
+const link = new HttpLink({
   uri: 'http://localhost:3000/graphql',
 
   // Use explicit `window.fetch` so tha outgoing requests
   // are captured and deferred until the Service Worker is ready.
   fetch: (...args) => fetch(...args),
```
Comment (atdrago) [inline examples/graphql-react-apollo/src/ApolloClient.js:10]: I noticed that changing this to `fetch: window.fetch` also worked, but I didn't want to make unnecessary changes. Just calling it out as a potential simplification
```suggestion
  fetch: window.fetch,
```

File: examples/graphql-react-apollo/src/ApolloClient.js [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
 })
+
+// Isolate Apollo client so it could be reused
+// in both application runtime and tests.
+export const client = new ApolloClient({
+  cache,
```
Comment (atdrago) [inline examples/graphql-react-apollo/src/ApolloClient.js:16]: `cache` is a required property, so a I created a simple `InMemoryCache` and passed it in

File: examples/graphql-react-apollo/src/ApolloClient.js (hunk 3 above)
Comment (atdrago) [inline examples/graphql-react-apollo/src/ApolloClient.js:10]: We need to create an `HttpLink` so we can pass in `fetch`, so I also passed in `uri` here. Then `link` gets passed into `ApolloClient`. This matches Apollo's advanced HTTP guide https://www.apollographql.com/docs/react/networking/advanced-http-networking/#customizing-request-logic

**Review by kettanaito — APPROVED** [review]
Hi, @atdrago! Thank you for preparing this tremendous update!

It looks great, I'm stunned by your precise approach on this and explanations following your changes. I'll give it a local run and it'll be ready to go.

File: examples/graphql-react-apollo/src/ApolloClient.js (hunk 3 above)
Comment (atdrago) [reply examples/graphql-react-apollo/src/ApolloClient.js:10]: Good to know! Happy to leave it as is. 👍

**Comment by kettanaito** [comment]
I confirm the example as working. Great job!
